### PR TITLE
Fix unauthorized error when saving workflows

### LINKED ISSUE #5095: [CAL-195] No longer able to save workflow after selecting event type
### Issue Summary

I no longer can add new workflows in my account. looks like this only happens when I select an event type in the workflow.

I get the following error in the dev console:

```
invalid/:1          Failed to load resource: net::ERR_FAILED
auth/login:1 The resource https://app.cal.com/cal.ttf was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
main-e5b513407ef9dae7.js:1 Error: Abort fetching component for route: "/event-types"
    at main-e5b513407ef9dae7.js:1:56821
    at main-e5b513407ef9dae7.js:1:69622
    at l (main-e5b513407ef9dae7.js:1:109170)
    at Generator._invoke (main-e5b513407ef9dae7.js:1:108958)
    at forEach.e.<computed> [as next] (main-e5b513407ef9dae7.js:1:109593)
    at r (main-e5b513407ef9dae7.js:1:880)
    at u (main-e5b513407ef9dae7.js:1:1085)
(anonymous) @ main-e5b513407ef9dae7.js:1
event-types:1 The resource https://app.cal.com/cal.ttf was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
DevTools failed to load source map: Could not load content for chrome-extension://kghbmcgihmefcbjlfiafjcigdcbmecbf/static/js/content.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
workflows/2857:1 The resource https://app.cal.com/cal.ttf was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
_app-8e391c55f39f7233.js:1          POST https://app.cal.com/api/trpc/viewer.workflows.update?batch=1 401
(anonymous) @ _app-8e391c55f39f7233.js:1
Promise.then (async)
promise @ _app-8e391c55f39f7233.js:1
c @ _app-8e391c55f39f7233.js:1
fetch @ _app-8e391c55f39f7233.js:1
r @ _app-8e391c55f39f7233.js:1
setTimeout (async)
load @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
promise @ _app-8e391c55f39f7233.js:1
u @ _app-8e391c55f39f7233.js:1
requestAsPromise @ _app-8e391c55f39f7233.js:1
mutation @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
fn @ _app-8e391c55f39f7233.js:1
m @ _app-8e391c55f39f7233.js:1
l @ _app-8e391c55f39f7233.js:1
t @ _app-8e391c55f39f7233.js:1
execute @ _app-8e391c55f39f7233.js:1
await in execute (async)
mutate @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
(anonymous) @ [workflow]-52d09e7142b08d8a.js:1
l @ main-e5b513407ef9dae7.js:1
(anonymous) @ main-e5b513407ef9dae7.js:1
forEach.e.<computed> @ main-e5b513407ef9dae7.js:1
o @ 3839-0b02b9020ea0924e.js:1
s @ 3839-0b02b9020ea0924e.js:1
(anonymous) @ 3839-0b02b9020ea0924e.js:1
(anonymous) @ 3839-0b02b9020ea0924e.js:1
(anonymous) @ [workflow]-52d09e7142b08d8a.js:1
(anonymous) @ 3955-531b0ff2227a6e98.js:1
await in (anonymous) (async)
onSubmit @ 9762-ee5cf6526001be9f.js:1
Oe @ framework-a4238fd52304273f.js:1
Ae @ framework-a4238fd52304273f.js:1
(anonymous) @ framework-a4238fd52304273f.js:1
Ir @ framework-a4238fd52304273f.js:1
Ur @ framework-a4238fd52304273f.js:1
(anonymous) @ framework-a4238fd52304273f.js:1
ss @ framework-a4238fd52304273f.js:1
Le @ framework-a4238fd52304273f.js:1
Hr @ framework-a4238fd52304273f.js:1
Kn @ framework-a4238fd52304273f.js:1
Qn @ framework-a4238fd52304273f.js:1
_app-8e391c55f39f7233.js:1  << mutation #28 viewer.workflows.update Object
(anonymous) @ _app-8e391c55f39f7233.js:1
s @ _app-8e391c55f39f7233.js:1
error @ _app-8e391c55f39f7233.js:1
error @ _app-8e391c55f39f7233.js:1
error @ _app-8e391c55f39f7233.js:1
error @ _app-8e391c55f39f7233.js:1
error @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
Promise.then (async)
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
subscribe @ _app-8e391c55f39f7233.js:1
promise @ _app-8e391c55f39f7233.js:1
u @ _app-8e391c55f39f7233.js:1
requestAsPromise @ _app-8e391c55f39f7233.js:1
mutation @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
fn @ _app-8e391c55f39f7233.js:1
m @ _app-8e391c55f39f7233.js:1
l @ _app-8e391c55f39f7233.js:1
t @ _app-8e391c55f39f7233.js:1
execute @ _app-8e391c55f39f7233.js:1
await in execute (async)
mutate @ _app-8e391c55f39f7233.js:1
(anonymous) @ _app-8e391c55f39f7233.js:1
(anonymous) @ [workflow]-52d09e7142b08d8a.js:1
l @ main-e5b513407ef9dae7.js:1
(anonymous) @ main-e5b513407ef9dae7.js:1
forEach.e.<computed> @ main-e5b513407ef9dae7.js:1
o @ 3839-0b02b9020ea0924e.js:1
s @ 3839-0b02b9020ea0924e.js:1`
(anonymous) @ 3839-0b02b9020ea0924e.js:1
(anonymous) @ 3839-0b02b9020ea0924e.js:1
(anonymous) @ [workflow]-52d09e7142b08d8a.js:1
(anonymous) @ 3955-531b0ff2227a6e98.js:1
await in (anonymous) (async)
onSubmit @ 9762-ee5cf6526001be9f.js:1
Oe @ framework-a4238fd52304273f.js:1
Ae @ framework-a4238fd52304273f.js:1
(anonymous) @ framework-a4238fd52304273f.js:1
Ir @ framework-a4238fd52304273f.js:1
Ur @ framework-a4238fd52304273f.js:1
(anonymous) @ framework-a4238fd52304273f.js:1
ss @ framework-a4238fd52304273f.js:1
Le @ framework-a4238fd52304273f.js:1
Hr @ framework-a4238fd52304273f.js:1
Kn @ framework-a4238fd52304273f.js:1
Qn @ framework-a4238fd52304273f.js:1
workflows/2857:1 The resource https://app.cal.com/cal.ttf was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
```

### Steps to Reproduce

1. Went to workflows
2. Add new workflow
3. select event type
4. saving does not work.

### Technical details

I am using [cal.com](http://cal.com) cloud.

> From [Linear-GitHub Sync](https://synclinear.com)

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Fixes issues that unauthorized error was thrown when adding an event type to workflows that doesn't have an userId or teamId and only has the user saved in users[]. 

Fixes #5095

**Environment**: Staging(main branch)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/trpc/server/routers/viewer/workflows.tsx [hunk 1]
```diff
@@ -296,7 +297,8 @@ export const workflowsRouter = createProtectedRouter()
         if (
           newEventType &&
           newEventType.userId !== user.id &&
-          !newEventType?.team?.members.filter((membership) => membership.userId === user.id).length
+          !newEventType?.team?.members.filter((membership) => membership.userId === user.id).length &&
+          !newEventType?.users.filter((eventTypeUser) => eventTypeUser.id === user.id).length
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/workflows.tsx:301]:
```suggestion
          !newEventType?.team?.members.find((membership) => membership.userId === user.id) &&
          !newEventType?.users.find((eventTypeUser) => eventTypeUser.id === user.id)
```

**Review by alannnc — CHANGES_REQUESTED** [review]
Should be ready to go after changes are made

**Review by hariombalhara — APPROVED** [review]

File: packages/trpc/server/routers/viewer/workflows.tsx [hunk 2]
```diff
@@ -296,7 +297,8 @@ export const workflowsRouter = createProtectedRouter()
         if (
           newEventType &&
           newEventType.userId !== user.id &&
-          !newEventType?.team?.members.filter((membership) => membership.userId === user.id).length
+          !newEventType?.team?.members.find((membership) => membership.userId === user.id) &&
+          !newEventType?.users.find((eventTypeUser) => eventTypeUser.id === user.id)
         ) {
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/workflows.tsx:302]: It definitely deserves a utility fn that can be called like `canMakeChangesToEvent(eventType, user)`. I am sure we must be using similar authorization checks elsewhere. So, either similar fn should exist already or we should make it.

Comment (alannnc) [reply packages/trpc/server/routers/viewer/workflows.tsx:302]: Also split each conditional in individual functions and reuse. It's a NIT change but it's welcome.
isUserIdMemberIn

**Review by alannnc — APPROVED** [review]

**Pull request merged**